### ISSUE TITLE
Hyperion .NET Core to .NET Full framework

### ISSUE THREAD

**Issue by leo12chandu** [akkadotnet/akka.net#3587]
When creating a new issue, please make sure the following information is part of your issue description. (if applicable). Thank You!

- Which Akka.Net version you are using
1.3.8
- On which platform you are using Akka.Net
Windows, VS2017
- A list of steps to reproduce the issue. Or an gist or github repo which can be easily used to reproduce your case.
When we have akka.net app that is hosted in .NET Core but may have clients that are running in .NET Full framework, there is compatibility issues with serialization.

**Solution:**-
I have an implementation for Hyperion to work across .Net Core to .NET Full framework. Currently it supports collections that switch between mscorlib to System.Private.CorLib as well as ExpandoObjects going from System.Dynamic to System.Linq.

It can be found at https://github.com/leo12chandu/akka.net/blob/dev/src/core/Akka/Serialization/HyperionByteHelper.cs

**Usage**:-
Within the FromBinary of your Akka.Serialization.Serializer custom serializer class, call 

```
//Cross Framework compatible conversions so if client and server use .Net Core and Fullframework. serializer is compatible.
            bytes = HyperionByteHelper.GetBytesForCurrentFramework(bytes);
```
 
The ToBinary can be left as-is

**Comment by at-zia** [comment]
Hi @leo12chandu, thanks for this. However I'm unable to compile the code, due to missing reference to the bytes helper (eg. `bytes.Locate()`). I couldn't find the definition in the repo itself. Can you please check?

**Comment by Aaronontheweb** [comment]
This has been implemented as of Hyperion v0.9.10 https://github.com/akkadotnet/Hyperion/releases/tag/0.9.10
